### PR TITLE
[WIP] - NPEs cleanup

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Just a cleanup of potential NPEs in the core

### CODE REVIEW

File: core/src/main/java/hudson/util/XStream2.java [hunk 1]
```diff
@@ -434,4 +463,4 @@ public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext co
 
     }
 
-}
+}
```
Comment (KostyaSha) [inline core/src/main/java/hudson/util/XStream2.java]: don't remove new line

**Comment by daniel-beck** [comment]
@oleg-nenashev Are you still interested in this PR? FWIW I think multiple small PRs in this area are preferable to one larger one.

**Comment by oleg-nenashev** [comment]
@daniel-beck 
I'm still interested, but I agree that mini-PRs are better.
I hope to crunch it a bit once I finish the main work. Makes sense to have more annotations in 2.x.1 LTS

**Comment by oleg-nenashev** [comment]
At some point I'll submit as a series of small PRs...

**Pull request closed without merging**